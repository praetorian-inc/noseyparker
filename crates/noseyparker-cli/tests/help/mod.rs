//! Tests for Nosey Parker `help` functionality

use super::*;

#[test]
fn no_args() {
    assert_cmd_snapshot!(noseyparker!().assert().failure());
}

#[test]
fn help() {
    assert_cmd_snapshot!(noseyparker_success!("help"));
}

#[test]
fn help_short() {
    assert_cmd_snapshot!(noseyparker_success!("-h"));
}

#[test]
fn help_scan() {
    with_settings!({
        filters => vec![
            (r"(?m)(scanning threads\s+)\[default: \d+\]", r"$1[default: DEFAULT]"),
        ],
    }, {
        assert_cmd_snapshot!(noseyparker_success!("help", "scan"));
    });
}

#[test]
fn help_scan_short() {
    with_settings!({
        filters => vec![
            (r"(?m)(scanning threads\s+)\[default: \d+\]", r"$1[default: DEFAULT]"),
        ],
    }, {
        assert_cmd_snapshot!(noseyparker_success!("scan", "-h"));
    });
}

#[test]
fn help_summarize() {
    assert_cmd_snapshot!(noseyparker_success!("help", "summarize"));
}

#[test]
fn help_summarize_short() {
    assert_cmd_snapshot!(noseyparker_success!("summarize", "-h"));
}

#[test]
fn help_report() {
    with_settings!({
        filters => vec![
            (r"(?m)(denoising threads when using the CPU\s+)\[default: \d+\]", r"$1[default: DEFAULT]"),
        ],
    }, {
        assert_cmd_snapshot!(noseyparker_success!("help", "report"));
    });
}

#[test]
fn help_report_short() {
    with_settings!({
        filters => vec![
            (r"(?m)(denoising threads when using the CPU\s+)\[default: \d+\]", r"$1[default: DEFAULT]"),
        ],
    }, {
        assert_cmd_snapshot!(noseyparker_success!("report", "-h"));
    });
}

#[test]
fn help_datastore() {
    assert_cmd_snapshot!(noseyparker_success!("help", "datastore"));
}

#[test]
fn help_rules() {
    assert_cmd_snapshot!(noseyparker_success!("help", "rules"));
}

#[test]
fn help_github() {
    assert_cmd_snapshot!(noseyparker_success!("help", "github"));
}

#[test]
fn help_github_short() {
    assert_cmd_snapshot!(noseyparker_success!("github", "-h"));
}

#[test]
fn help_github_repos() {
    assert_cmd_snapshot!(noseyparker_success!("help", "github", "repos"));
}

#[test]
fn help_github_repos_short() {
    assert_cmd_snapshot!(noseyparker_success!("github", "repos", "-h"));
}

// #[test]
// fn version_short() {
//     assert_cmd_snapshot!(noseyparker_success!("-V"));
// }

#[test]
fn version_long() {
    with_settings!({
        filters => vec![
            (r"(?m)^(    [^:]+:[ \t]+).*$", r"$1<PLACEHOLDER>")
        ],
    }, {
        assert_cmd_snapshot!(noseyparker_success!("--version"));
    });
}

#[test]
fn version_command() {
    // there is no `version` command
    assert_cmd_snapshot!(noseyparker_failure!("version"));
}
