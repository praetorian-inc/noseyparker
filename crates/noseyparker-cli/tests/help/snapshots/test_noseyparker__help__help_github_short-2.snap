---
source: crates/noseyparker-cli/tests/help/mod.rs
expression: stdout
---
Interact with GitHub

Usage: noseyparker github [OPTIONS] <COMMAND>

Commands:
  repos  Interact with GitHub repositories
  help   Print this message or the help of the given subcommand(s)

Options:
      --github-api-url <URL>  Use the given URL for GitHub API access [default:
                              https://api.github.com/] [aliases: api-url]
  -h, --help                  Print help (see more with '--help')

Global Options:
  -v, --verbose...       Enable verbose output
      --color <MODE>     Enable or disable colored output [default: auto] [possible values: auto,
                         never, always]
      --progress <MODE>  Enable or disable progress bars [default: auto] [possible values: auto,
                         never, always]

